src/v2/view-builder/views/EnrollProfileView.js

```diff
@@ -61,8 +61,20 @@ export default BaseView.extend({
   Body,
   Footer,
   createModelClass(currentViewState, optionUiSchemaConfig, settings) {
-    let ModelClass = BaseView.prototype.createModelClass.apply(this, arguments);
     const currentSchema = JSON.parse(JSON.stringify((currentViewState.uiSchema)));
+    const appState = this.options.appState;
+    let ModelClass = BaseView.prototype.createModelClass.apply(this, arguments, appState);
+
+    ModelClass = ModelClass.extend({
+      toJSON: function() {
+        const modelJSON = Model.prototype.toJSON.call(this, arguments, appState);
+        // remove unwanted data from modelJSON
+        if(modelJSON.userProfile) {
+          appState.cleanViewModelState(modelJSON.userProfile);
```

Comment:
we don't need `cleanViewModelState` anymore if we are overriding `toJSON`.. we can simply just delete empty optional attributes.

if we are adding `userProfile` in `cleanViewModelState` then this function is not generalized and not serving the purpose of keeping it in AppState.js